pygeoapi/provider/geojson.py

```diff
@@ -130,6 +131,7 @@ def _load(self, skip_geometry=None, properties=[], select_properties=[]):
                                    if k in set(self.properties) | set(select_properties)}  # noqa
         return data
 
+    @crs_transform
```

Comment:
Does this decorate mean an extra hop for all provider `query` and `get` functions?

Reply:
The decorator iterates through the `features` returned by the providers' `query` and `get` functions, transform the geometries' coordinates if needed (meaning that `crs_transform_spec != None` in `query()` and `get()` calls in `api.py`) before the `Feature`/`FeatureCollection` is assigned to the `content` variable in `api.py`. The decorator approach was used to avoid code duplication (fewer lines of code if the transformation logic is taken out of the providers' implementation), for simplicity and for supporting coordinates transformation for all providers. `PostgreSQLProvider` has its own implementation and does not use the `@crs_transform` decorator. The implementation for the `PostgreSQL` provider is more optimized as it does not re-iterate through the returned `features`, and we may want to optimize the coordinates transformation implementation of other providers later, especially if these providers have *native support* for coordinates transformation. In practice, re-iterating through some thousands of features would not result in much overhead.

Reply:
Yes, agree. Also the OGRProvider does its own transforms. Our plan is to first have #1174 merged, having a stable version, CITE and INSPIRE compliant. Then do refactoring in separate PRs:

- move all CRS-dedicated/related functionality and constants spread over `util.py` and `api.py`  into a new module `pygeoapi/crs.py`
- enhance `CrsTransformSpec` with direct-to-call transform functions as `partials` (e.g. for Shapely and GeoJSON dicts) that Providers without native CRS-support can call in their Feature-Collection loop
- upgrade Providers to using the CrsTransformSpec-based transforms, removing the Decorator, maybe even completely